.ci/test-project/packages/magictestlib_cached/package.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import glob
+import os
+import shutil
+import socket
+from os import environ as env
+
+import llnl.util.tty as tty
+
+from spack import *
+
+def get_spec_path(spec, package_name, path_replacements={}, use_bin=False):
+    """Extracts the prefix path for the given spack package
+       path_replacements is a dictionary with string replacements for the path.
+    """
+
+    if not use_bin:
+        path = spec[package_name].prefix
+    else:
+        path = spec[package_name].prefix.bin
+
+    path = os.path.realpath(path)
+
+    for key in path_replacements:
+        path = path.replace(key, path_replacements[key])
+
+    return path
+
+
```

Comment:
```suggestion
```

This shouldn't be needed anymore.

Reply:
its used here. maybe i didn't setup the `initconfig_package_entries` function correctly https://github.com/LLNL/uberenv/blob/5b014f91f891bfd96570ebc02769e9088506c63e/.ci/test-project/packages/magictestlib_cached/package.py#L93

Reply:
See above comment about reducing complexity and removing all that logic.